csharp/ql/src/semmle/code/csharp/dataflow/ExternalFlow.qll

```diff
@@ -328,30 +328,49 @@ private predicate elementSpec(
 private class UnboundValueOrRefType extends ValueOrRefType {
   UnboundValueOrRefType() { this.isUnboundDeclaration() }
 
-  UnboundValueOrRefType getASubTypeUnbound() { result = this.getASubType().getUnboundDeclaration() }
+  UnboundValueOrRefType getASubTypeUnbound() {
+    exists(Type t |
+      result.getABaseType() = t and
+      this = t.getUnboundDeclaration()
+    )
+  }
+}
+
+private class UnboundCallable extends Callable, Virtualizable {
+  UnboundCallable() { this.isUnboundDeclaration() }
+
+  predicate overridesOrImplementsUnbound(UnboundCallable that) {
+    exists(Callable c |
+      this.overridesOrImplementsOrEquals(c) and
+      this != c and
+      that = c.getUnboundDeclaration()
+    )
+  }
 }
 
 bindingset[namespace, type, subtypes]
 private UnboundValueOrRefType interpretType(string namespace, string type, boolean subtypes) {
-  exists(UnboundValueOrRefType t |
-    t.hasQualifiedName(namespace, type) and
-    if subtypes = true then result = t.getASubTypeUnbound*() else result = t
+  exists(UnboundValueOrRefType t | t.hasQualifiedName(namespace, type) |
+    t = result
+    or
+    subtypes = true and
+    result = t.getASubTypeUnbound+()
```

Comment:
Shouldn't this disjunct be removed?

Reply:
It is needed for `InterpretedCallable` (which is a mere performance optimization, used to restrict the callables to construct signature strings for).